modules/apps/29-fee/ibc_middleware.go

```diff
@@ -267,10 +267,26 @@ func (im IBCMiddleware) OnAcknowledgementPacket(
 
 	packetID := channeltypes.NewPacketId(packet.SourcePort, packet.SourceChannel, packet.Sequence)
 	feesInEscrow, found := im.keeper.GetFeesInEscrow(ctx, packetID)
-	if found {
+	if !found {
+		// call underlying callback
+		return im.app.OnAcknowledgementPacket(ctx, packet, ack.AppAcknowledgement, relayer)
+	}
+
+	payee, found := im.keeper.GetPayeeAddress(ctx, relayer.String(), packet.SourceChannel)
+	if !found {
 		im.keeper.DistributePacketFeesOnAcknowledgement(ctx, ack.ForwardRelayerAddress, relayer, feesInEscrow.PacketFees, packetID)
+
+		// call underlying callback
+		return im.app.OnAcknowledgementPacket(ctx, packet, ack.AppAcknowledgement, relayer)
 	}
 
+	payeeAddr, err := sdk.AccAddressFromBech32(payee)
+	if err != nil {
+		return sdkerrors.Wrapf(err, "failed to create sdk.Address from payee: %s", payee)
```

Comment:
If an error is returned here, then no fees are paid at all, not even to the forward relayer that submitted the recv message. Shouldn't we try at least to distribute fees to the forward relayer? Otherwise we are penalizing the forward relayer for a mistake on setting the payee (which is caused by potentially a completely different relayer operator). Maybe we could also log or emit an event with the error.

Reply:
Or maybe we could just pay the fees to the `relayer` address in this case. At least somebody will get the fees!

Reply:
Ah, I see now [this comment](https://github.com/cosmos/ibc-go/pull/1526#discussion_r895841316)...

Reply:
Yep, was going to link the comment. I also think it would make more sense to distribute the fees atomically either way. In the more or less unlikely event of an error I think it makes most sense to just override the payee address by submitting the tx again and attempt to finish the packet lifecycle and distribute all fees at once